shell/common/display_manager.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_COMMON_DISPLAY_MANAGER_H_
+#define FLUTTER_SHELL_COMMON_DISPLAY_MANAGER_H_
+
+#include <mutex>
+#include <vector>
+
+#include "flutter/shell/common/display.h"
+
+namespace flutter {
+
+/// The update type parameter that is passed to
+/// `DisplayManager::HandleDisplayUpdates`.
+enum class DisplayUpdateType {
+  /// `flutter::Display`s that were active during start-up. A display is
+  /// considered active if:
+  ///    1. The frame buffer hardware is connected.
+  ///    2. The display is drawable, e.g. it isn't being mirrored from another
+  ///    connected display or sleeping.
+  kStartup
+};
+
+/// Manages lifecycle of the connected displays. This class is thread-safe.
+class DisplayManager {
```

Comment:
How will this communicate display information to the `PlatformConfiguration` class for exposure in `dart:ui`?

For instance, it currently holds the list of Windows (right now, only one, but eventually more), and I assume that display information will be useful to be there too, so it can be the source of truth for `dart:ui` calls to get information about the displays.

Reply:
I wasn't aware of `PlatformConfiguration`, but `PlatformConfigurationClient` can be provided a reference to `DisplayManager` via `Shell`. I haven't spent a lot of time looking at it, but seems like it should be possible as `engine` seems to be handling some of the tasks there via delegation.